topic-operator/src/main/java/io/strimzi/operator/topic/cruisecontrol/CruiseControlClient.java

```diff
@@ -0,0 +1,192 @@
+     * @param rackEnabled Whether rack awareness is enabled.
+     * @param sslEnabled Whether SSL is enabled.
+     * @param sslCertificate SSL certificate.
+     * @param authEnabled Whether authentication is enabled.
+     * @param authUsername Authentication username.
+     * @param authPassword Authentication password.
+     * @return Cruise Control client.
+     */
+    static CruiseControlClient create(String serverHostname,
+                                      int serverPort,
+                                      boolean rackEnabled,
+                                      boolean sslEnabled,
+                                      byte[] sslCertificate,
+                                      boolean authEnabled,
+                                      String authUsername,
+                                      String authPassword) {
+        return new CruiseControlClientImpl(
+            serverHostname,
+            serverPort,
+            rackEnabled,
+            sslEnabled,
+            sslCertificate,
+            authEnabled,
+            authUsername,
+            authPassword
+        );
+    }
+
+    /**
+     * Send a POST request to {@code topic_configuration} endpoint.
+     * This can be used to request replication factor changes (async operation).
+     * 
+     * @param kafkaTopics List of Kafka topics.
+     * @return The user task id.
+     */
+    String topicConfiguration(List<KafkaTopic> kafkaTopics);
+
+    /**
+     * Send a GET request to {@code user_tasks} endpoint.
+     * This can be used to check a task execution result (sync operation).
+     * 
+     * @param userTaskIds Set of user task ids.
+     * @return User tasks response.
+     */
+    UserTasksResponse userTasks(Set<String> userTaskIds);
+
+    /**
+     * Get the error message from HTTP response.
+     * 
+     * @param response The HTTP response.
+     * @return The error message.
+     */
+    Optional<String> errorMessage(HttpResponse<String> response);
+    
+    /**
+     * Topic names grouped by replication factor value.
+     * 
+     * @param topicByReplicationFactor Topic names grouped by replication factor value.
+     */
+    record ReplicationFactor(@JsonProperty("topic_by_replication_factor") Map<Integer, String> topicByReplicationFactor) { }
+
+    /**
+     * Replication factor changes.
+     * 
+     * @param replicationFactor Replication factor value.
+     */
+    record ReplicationFactorChanges(@JsonProperty("replication_factor") ReplicationFactor replicationFactor) { }
+    
+    /**
+     * The user task.
+     * 
+     * @param status Status.
+     * @param clientIdentity Client identity.
+     * @param requestURL Request URL.
+     * @param userTaskId User task id.
+     * @param startMs Start time in ms.
+     */
+    record UserTask(
+        @JsonProperty("Status") String status,
+        @JsonProperty("ClientIdentity") String clientIdentity,
+        @JsonProperty("RequestURL") String requestURL,
```

Comment:
`java.net.URI` would be more typesafe.

Reply:
Yeah, but the problem here is that it only returns the path and query string.